### PR TITLE
Cryptographically secure pseudo-random numbers!

### PR DESCRIPTION
**Pull request by turian**
With unit tests, so the results are replicable.

We should profile this but I still think pow and sin will be the biggest culprits.

### CODE REVIEW

File: torchsynth/synth.py [hunk 1]
```diff
@@ -51,68 +54,90 @@ def buffer_size(self) -> T:
...
-                "note_on": NoteOnButton(synthglobals),
-                "pitch_adsr": ADSR(synthglobals),
-                "amp_adsr": ADSR(synthglobals),
-                "vco_1": SineVCO(synthglobals),
-                "vco_2": SquareSawVCO(synthglobals),
-                "noise": Noise(synthglobals),
-                "vca": VCA(synthglobals),
-                "vca_ratio": CrossfadeKnob(synthglobals),
-            }
+            [
```
Comment (turian) [inline torchsynth/synth.py:128]: This is a list now to ensure all the modules are in deterministic order

File: torchsynth/module.py [hunk 2]
```diff
@@ -354,7 +354,12 @@ def __init__(
         # TODO: Make sure this is on GPU
         # https://pytorch-lightning.readthedocs.io/en/latest/advanced/multi_gpu.html#init-tensors-using-type-as-and-register-buffer
         # Do we want to detach clone?
-        self.phase = self.get_parameter("initial_phase").detach().clone()
+        # Do we want this persistent?
+        # https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.register_buffer
```
Comment (jorshi) [inline torchsynth/module.py:358]: Commented on this in other PR, but I don't think we need it persistent. If it is then it basically allows us to save and load a synth a keep these phase values right?

File: torchsynth/synth.py [hunk 3]
```diff
@@ -51,68 +54,91 @@ def buffer_size(self) -> T:
...
+            self.randomize(seed=batch_idx)
+        return self._forward(*args, **kwargs)
+
+    def randomize(self, seed: Optional[int]):
+        """
+        Randomize all parameters
+        """
+        if seed:
+            # Profile to make sure this isn't too slow?
+            mt19937_gen = csprng.create_mt19937_generator(seed)
```
Comment (jorshi) [inline torchsynth/synth.py:111]: cool -- I like that for extracting specific batches in a deterministic way

**Review by jorshi — APPROVED** [review]

File: examples/lightningsynth.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (turian) [reply examples/lightningsynth.py:1]: maybe not we're still getting lightning working

File: torchsynth/module.py (hunk 2 above)
Comment (turian) [reply torchsynth/module.py:358]: Okay make a PR to fix?

Comment (jorshi) [reply torchsynth/module.py:358]: Will do

**Pull request merged**